src/models/BibDetails.ts

```diff
@@ -362,6 +364,36 @@ export default class BibDetails {
   }
 
   buildSubjectHeadings(): SubjectHeadingDetail {
+    if (!this.bib.subjectHeadings) return
```

Comment:
`buildSubjectHeadings` and `buildSubjectLiterals` have a lot of duplicated logic. It looks like the main logical difference is the check for uuid on line 369, and otherwise the split is which key it's mapping over and the url string. Can the duplicated logic be extracted? 
Also, I think it would add some clarity to include `"SHEP"` in the subject headings function name, as well as include `"links"` or `"urls"` in both function names.

Reply:
These are good suggestions but I would like to do these refinements after the release since they don't affect functionality.